File: document/cgi/show/file_list_util.pl
use strict;
use warnings;

sub makeHTMLFileString {
    my ($fileName) = @_;
    return qq(<li><a href="/show/file.py?file=$fileName">$fileName</a></li>\n);
}

sub getHTMLStringList {
    my (@fileList) = @_;
    my $htmlStringList = '';
    foreach my $file (@fileList) {
        $htmlStringList .= makeHTMLFileString($file);
    }
    return $htmlStringList;
}

1;

File: document/cgi/show/file_list_util.py
def makeHTMLFileString(fileName):
	htmlFileString = '<li><a href="/show/file.py?file={file}">{file}</a></li>'.format(
		file = fileName)
	return (htmlFileString)

def getHTMLStringList(fileList):
	htmlStringList = ''
	for file in fileList:
		htmlStringList += makeHTMLFileString(file)
	return (htmlStringList)
